### PR TITLE
FEATURE: TNT-1152 PV GeoChart supports Tiger

### PR DESCRIPTION
**Pull request by ivan-nejezchleb**
---

Supported PR commands:

| Command                                               | Description                                                |
| ----------------------------------------------------- | ---------------------------------------------------------- |
| `ok to test`                                          | Re-run standard checks                                     |
| `extended check sonar`                                | SonarQube tests                                            |
| `extended check plugins`                              | Dashboard plugins tests                                    |
| `extended test - backstop`                            | BackstopJS tests                                           |
| **E2E Cypress tests commands - TIGER**                |                                                            |
| `extended test - tiger-cypress - isolated <testName>` | Run isolated tests running against recorded Tiger backend. |
| `extended test - tiger-cypress - record <testName>`   | Create a new recording for isolated Tiger tests.           |
| **E2E Cypress tests commands - BEAR**                 |                                                            |
| `extended test - cypress - isolated <testName>`       | Run isolated tests running against recorded Bear backend.  |
| `extended test - cypress - record <testName>`         | Create a new recording for isolated Bear tests.            |

`<testName>` in cypress commands is used to filter specfiles. Example, to run record with BEAR backend

-   Against `dashboard.spec.ts` and `drilling.spec.ts`, execute command `extended test - cypress - record dashboard,drilling`
-   Against all specfiles, execute command `extended test - cypress - record` or `extended test - cypress - record *`

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `extended test - backstop` passes
-   [ ] `extended test - tiger-cypress - record` to record new mapping files (Tiger BE)
-   [ ] `extended test - cypress - record` to record new mapping files (Bear BE)
-   [ ] `extended test - tiger-cypress - isolated` passes
-   [ ] `extended test - cypress - isolated` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/11088/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/13181/ : FAILURE in 24m 52s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/11090/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/13183/ : SUCCESS in 19m 46s

File: libs/sdk-backend-tiger/src/backend/workspace/insights/mocks/visualizationClasses.ts [hunk 1]
```diff
@@ -172,19 +172,18 @@ export const visualizationClasses: IVisualizationClass[] = sortBy(
                 uri: "/gdc/md/gtl83h4doozbp26q0kf5qg8uiyu4glyn/obj/1000",
             },
         },
-        // geochart must be disabled for now on tiger backend due to missing mapbox token infrastructure there (RAIL-3058)
```
Comment (kandl) [inline libs/sdk-backend-tiger/src/backend/workspace/insights/mocks/visualizationClasses.ts:175]: Commented code is equal to uncommented code on line 175-186. Should it be commented or not?

File: libs/sdk-backend-tiger/src/backend/workspace/catalog/datasetLoader.ts [hunk 2]
```diff
@@ -48,12 +49,12 @@ function getAttributeLabels(
...
-     *  using some indicator on the metadata object. for sakes of speed & after agreement with tiger team
-     *  falling back to use of id convention.
-     */
-    return label.id.search(/^.*\.geo__/) > -1;
+    const type = label.attributes?.valueType;
+
+    return (
+        type === JsonApiLabelOutAttributesValueTypeEnum.GEO_LATITUDE ||
+        type === JsonApiLabelOutAttributesValueTypeEnum.GEO_LONGITUDE
+    );
```
Comment (kandl) [inline libs/sdk-backend-tiger/src/backend/workspace/catalog/datasetLoader.ts:57]: Should we check also for GEO_PUSHPIN? or should it be removed from the LabelTypeConverter?

File: libs/sdk-model/api/sdk-model.api.md [hunk 3]
```diff
@@ -505,7 +513,7 @@ export interface IAttributeDescriptorBody {
 // @public
 export interface IAttributeDisplayFormMetadataObject extends IMetadataObject {
     attribute: ObjRef;
-    displayFormType?: string;
+    displayFormType?: AttributeDisplayFormType | string;
```
Comment (kandl) [inline libs/sdk-model/api/sdk-model.api.md:516]: String here is just for backward compatibility reason now (but it's not needed anymore),  is it correct? Or do we have still some display form types that are not covered by that enum?

File: libs/sdk-ui-dashboard/src/model/commandHandlers/widgets/validation/insightDrillDefinitionUtils.ts [hunk 4]
```diff
@@ -131,7 +132,7 @@ export interface InsightDrillDefinitionValidationData {
     availableDrillTargets: IAvailableDrillTargets;
 }
 
-export const hyperlinkDisplayFormType = "GDC.link";
+export const hyperlinkDisplayFormType = AttributeDisplayFormType.HYPERLINK;
```
Comment (kandl) [inline libs/sdk-ui-dashboard/src/model/commandHandlers/widgets/validation/insightDrillDefinitionUtils.ts:135]: This assignment seems redundant, can we use `AttributeDisplayFormType.HYPERLINK` directly?

File: libs/sdk-ui-ext/src/internal/components/pluggableVisualizations/geoChart/geoConfigCodeGenerator.ts [hunk 5]
```diff
@@ -42,6 +58,43 @@ export function geoConfigFromInsight(insight: IInsightDefinition, ctx?: IEmbeddi
...
+): IAttribute {
+    if (
+        bucketName === BucketNames.LOCATION &&
+        !ctx.backend?.capabilities.supportsSeparateLatitudeLongitudeLabels
+    ) {
+        const bucket = insightBucket(insight, bucketName);
+        return bucket && bucketAttribute(bucket);
+    }
+    // dont rely that Latitude is already in bucket, take both lat and long from properties
+    if (
```
Comment (kandl) [inline libs/sdk-ui-ext/src/internal/components/pluggableVisualizations/geoChart/geoConfigCodeGenerator.ts:87]: should it be else if ? combination of both seems unlikely

File: libs/sdk-backend-tiger/src/backend/workspace/insights/mocks/visualizationClasses.ts (hunk 1 above)
Comment (ivan-nejezchleb) [reply libs/sdk-backend-tiger/src/backend/workspace/insights/mocks/visualizationClasses.ts:175]: sry rebase artifact, will remove it

File: libs/sdk-model/api/sdk-model.api.md (hunk 3 above)
Comment (ivan-nejezchleb) [reply libs/sdk-model/api/sdk-model.api.md:516]: Bear model is using string type too. So not sure what surprises can come from MD to converter and need to be converted to this type. But main motivation was backward compatibility and I dont know any widely used type not covered by enum

File: libs/sdk-ui-ext/src/internal/components/pluggableVisualizations/geoChart/geoConfigCodeGenerator.ts (hunk 5 above)
Comment (ivan-nejezchleb) [reply libs/sdk-ui-ext/src/internal/components/pluggableVisualizations/geoChart/geoConfigCodeGenerator.ts:87]: fixed

File: libs/sdk-backend-tiger/src/backend/workspace/catalog/datasetLoader.ts (hunk 2 above)
Comment (ivan-nejezchleb) [reply libs/sdk-backend-tiger/src/backend/workspace/catalog/datasetLoader.ts:57]: Checking with Tiger team, if JsonApiLabelOutAttributesValueTypeEnum should not be without GEO_PUSHPIN. So far I will keep it everywhere

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/11101/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/13195/ : SUCCESS in 21m 12s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/11106/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/13200/ : SUCCESS in 21m 23s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/11108/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/13202/ : SUCCESS in 28m 02s

**Review by kandl — APPROVED** [review]
LGTM

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR3388

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul/250/ : FAILURE in 2m 01s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/3516/ : SUCCESS in 19s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/2365/ : SUCCESS in 16m 00s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/13205/ : SUCCESS in 21m 09s

**Pull request merged**